### PR TITLE
Generalized selector parsing and ptr verification

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
**Stack**:
- #1228
- #1226 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: crates/forge-runner/src/runtime.rs [hunk 1]
```diff
@@ -150,6 +195,19 @@ impl<Extension: ExtensionLogic> HintProcessorLogic for ExtendedRuntime<Extension
     }
 }
 
+impl<Handler, Runtime: SyscallPtrAccess + HintProcessor> SyscallPtrAccess
+    for RuntimeExtension<Handler, Runtime>
+{
+    fn get_mut_syscall_ptr(&mut self) -> &mut Relocatable {
+        let val: &mut _ = self.extended_runtime.get_mut_syscall_ptr();
```
Comment (Arcticae) [inline crates/forge-runner/src/runtime.rs:202]: why not just `&mut self.extended_runtime.get_mut_syscall_ptr()?`

**Review by Arcticae — APPROVED** [review]

File: crates/forge-runner/src/runtime.rs [hunk 2]
```diff
@@ -124,8 +159,18 @@ impl<Extension: ExtensionLogic> HintProcessorLogic for ExtendedRuntime<Extension
...
+            let system_ptr = get_ptr(vm, cell, &offset)?;
+
+            self.0.verify_syscall_ptr(system_ptr)?;
+
+            // We peek into memory to check the selector
+            let selector = SyscallSelector::try_from(felt_to_stark_felt(
+                &vm.get_integer(*self.0.get_mut_syscall_ptr()).unwrap(),
+            ))?;
+
             // TODO move selector parsing logic here
```
Comment (piotmag769) [inline crates/forge-runner/src/runtime.rs:172]: Isn't it done in this PR?

**Review by piotmag769 — APPROVED** [review]

File: crates/forge-runner/src/runtime.rs (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/runtime.rs:202]: Done

File: crates/forge-runner/src/runtime.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/runtime.rs]: Done

**Pull request merged**